Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -289,7 +289,9 @@ final class BuildPlanTests: XCTestCase {
 
         let barLinkArgs = try result.buildProduct(for: "Bar").linkArguments()
         XCTAssertEqual(barLinkArgs,
-            ["/fake/path/to/swiftc", "-g", "-L", "/path/to/build/debug", "-o", "/path/to/build/debug/libBar.\(PackageModel.Product.dynamicLibraryExtension)", "-module-name", "Bar", "-emit-library", "/path/to/build/debug/Bar.build/source.swift.o"])
+            ["/fake/path/to/swiftc", "-g", "-L", "/path/to/build/debug", "-o",
+            "/path/to/build/debug/libBar.\(try! Destination.hostDestination().dynamicLibraryExtension)",
```

Comment:
This is going to be expensive, we should make `hostDynamicLibraryExtension` static.